test/parallel/test-assert-async.js

```diff
@@ -29,7 +29,7 @@ assert.doesNotReject(() => {});
     type: assert.AssertionError,
     message: 'Failed'
   }));
-  assert.doesNotReject(() => promise);
+  assert.doesNotReject(() => promise).then(common.mustCall());
```

Comment:
Nit: use `common.crashOnUnhandledRejection()` at the top of the file instead of adding `then(common.mustCall())` everywhere.